tests/components/tessie/test_button.py

```diff
@@ -0,0 +1,27 @@
+"""Test the Tessie button platform."""
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN, SERVICE_PRESS
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+
+from .common import setup_platform
+
+
+async def test_buttons(hass: HomeAssistant) -> None:
+    """Tests that the buttons are correct."""
+
+    await setup_platform(hass)
+
+    # Test wake button
+    with patch(
+        "homeassistant.components.tessie.entity.TessieEntity.run",
```

Comment:
We shouldn't patch the integration, instead patch the library

Reply:
I have discussed this with @joostlek, we cannot figure out how to patch this library when its functions are passed through a DataClass. 

Patching `"homeassistant.components.tessie.button.wake"` results in
> pytest_socket.SocketBlockedError: A test tried to use socket.socket.

Reply:
You could patch the func in the dataclass

Reply:
But that won't work because its frozen...hmm

Reply:
```python
from contextlib import contextmanager

@contextmanager
def patch_description(key: str) -> AsyncMock:
    """Patch a description."""
    to_patch = next(filter(lambda x: x.key == key, DESCRIPTIONS))
    original = to_patch.func
    mock = AsyncMock()
    object.__setattr__(to_patch,"func",mock)
    yield mock
    object.__setattr__(to_patch,"func",original)
    
```

This will work to patch it


```python
    # Test wake button
    with patch_description("wake") as mock_wake:
        await hass.services.async_call(
            BUTTON_DOMAIN,
            SERVICE_PRESS,
            {ATTR_ENTITY_ID: ["button.test_wake"]},
            blocking=True,
        )
        await hass.async_block_till_done()
        mock_wake.assert_called_once()
```

Reply:
I have been hammering GitHub Copilot Chat to find a solution, and its also suggesting patching `TessieButtonEntityDescription`, but I havent been successful with that yet.

I have code coverage from other platforms that do not use Entity Descriptions, but any of mine that do, have this problem. So if we do find a fix I'll backport it to the tests already merged.

Reply:
Success, Ill raise a seperate PR to backport this into my other platforms.